File: 2021/day02A.js
const fs = require('fs')
const filename = process.argv[2] || 'day02_input.txt'

fs.readFile(filename, 'utf8', (err, data) => {
    if (err) throw err

    const lines = data.trim().split('\n')

    const commands = {forward: 0, up: 0, down: 0,}
    lines
        .map(line => line.split(' '))
        .reduce((cmds, curr) => {
            const [dir, amount] = curr
            cmds[dir] += Number.parseInt(amount, 10)

            return cmds
        }, commands)

    console.log((commands.down - commands.up) * commands.forward)
})

File: 2021/day02A.py
commands = {
    'forward': 0,
    'up': 0,
    'down': 0,
}

with open('day02_input.txt') as f:
    for line in f:
        cmd, arg = line.strip().split(' ')
        commands[cmd] += int(arg)

vertical = commands['down'] - commands['up']
horizontal = commands['forward']
print(f'Day 02\tPart A:\t{horizontal * vertical}')